terraform/lib/dependabot/terraform/file_parser.rb

```diff
@@ -109,6 +111,21 @@ def build_provider_dependency(file, name, details = {})
         )
       end
 
+      def deprecated_provider_error(file)
+        raise Dependabot::DependencyFileNotParseable.new(
+          file.path,
+          "This terraform provider syntax is now deprecated.\n"\
+          "See https://www.terraform.io/docs/language/providers/requirements.html "\
+          "for the new Terraform v0.13+ provider syntax."
+        )
+      end
+
+      def deprecated_provider?(details)
+        # The old syntax for terraform providers v0.12- looked like
+        # "tls ~> 2.1" which gets parsed as a string instead of a hash
+        !details.respond_to?(:fetch)
```

Comment:
Should we check if this actually uses the old style syntax, or at least if it's a string instead of checking it responds to fetch?